cdap-app-fabric/src/main/java/co/cask/cdap/scheduler/NoOpScheduler.java

```diff
@@ -99,6 +99,16 @@ public ProgramScheduleStatus getScheduleStatus(ScheduleId scheduleId) throws Not
   }
 
   @Override
+  public List<ProgramScheduleRecord> listScheduleRecords(ApplicationId appId) throws NotFoundException {
+    return null;
+  }
+
+  @Override
+  public List<ProgramScheduleRecord> listScheduleRecords(ProgramId programId) throws NotFoundException {
+    return null;
```

Comment:
in general better to return empty list rather than null so that client does not have to do the null checks.